### ISSUE TITLE
To request access to this beta feature, please contact us at stripe-terminal-betas@stripe.com.

### ISSUE THREAD

**Issue by Mike-Android-Chow** [stripe/stripe-terminal-android#369]
I can not connect the reader via usb and the error message is "To request access to this beta feature, please contact us at stripe-terminal-betas@stripe.com."  , But the bluetooth is fine.  Why does this happen?

**Comment by chr-stripe** [comment]
Hi @Mike-Android-Chow, can you share which version of the Terminal SDK you are using? USB connectivity was made generally available in version `2.22.0`, so we recommend developers use this version or later if you plan to leverage a USB-based connection to your mobile reader. 

Prior to `2.22.0` USB connectivity was still limited as a beta program, which might explain why you are seeing this message if you are using an older version from an account that was not part of the beta program.

**Comment by chr-stripe** [comment]
Closing due to inactivity.

**Comment by Mike-Android-Chow** [comment]
@chr-stripe I upgraded the sdk to 2.23.1 , but still see this error.  I guess the usb method is a beta feature, we can not use the usb method until we get the authorized from stripe?

**Comment by chr-stripe** [comment]
The code that emits that error message was removed in SDK version `2.22.0`. There is no manner in which you could be seeing that error message on that version or any version afterwards, including `2.23.1`.

Can you share the serial number of the reader you're trying to connect to over USB?

**Comment by Mike-Android-Chow** [comment]
> The code that emits that error message was removed in SDK version `2.22.0`. There is no manner in which you could be seeing that error message on that version or any version afterwards, including `2.23.1`.
> 
> Can you share the serial number of the reader you're trying to connect to over USB?

I will confirm it again in 2.23.1 .
the sn is STRM26138019778

**Comment by chr-stripe** [comment]
I can see there were several different SDK versions connecting to that reader, some with `2.21.1` (which would require access to this beta feature), and some from `2.23.1`, which worked successfully as expected. These requests all came from the same development test account, so I've added that account to the beta feature access.

I would still recommend you upgrade to a newer version of the SDK to use USB connections, since we have added several bug fixes and improvements to the USB connection feature since making it generally available.
